src/components/ScenarioParameters/components/EditModeButton.js

```diff
@@ -2,44 +2,56 @@
 // Licensed under the MIT license.
 
 import React from 'react';
+import { useSelector } from 'react-redux';
 import PropTypes from 'prop-types';
 import { Fade, Grid, IconButton, Tooltip } from '@material-ui/core';
 import PlayCircleOutlineIcon from '@material-ui/icons/PlayCircleOutline';
 import BackspaceIcon from '@material-ui/icons/Backspace';
 import { useTranslation } from 'react-i18next';
+import { ACL_PERMISSIONS } from '../../../services/config/accessControl';
+import { PermissionsGate } from '@cosmotech/ui';
 
 const EditModeButton = ({ classes, handleClickOnDiscardChange, handleClickOnUpdateAndLaunchScenario }) => {
   const { t } = useTranslation();
+  const userAppPermissions = useSelector((state) => state.auth.permissions);
+  const currentScenario = useSelector((state) => state.scenario.current.data);
+  const userPermissionsOnCurrentScenario = currentScenario?.security?.currentUserPermissions || [];
+  const userAppAndScenarioPermissions = userAppPermissions.concat(userPermissionsOnCurrentScenario);
```

Comment:
You can mutualize this in a new hook file, maybe in a hooks/UserPermissionsHooks ?
Be carreful to use useMemo here because the ref change in each refresh